examples/CRISP/contracts/CRISPPolicy.sol

```diff
@@ -2,51 +2,85 @@
 pragma solidity >=0.8.27;
 
 import {IEnclavePolicy} from "@gnosis-guild/enclave/contracts/interfaces/IEnclavePolicy.sol";
-import {AdvancedPolicy} from "@excubiae/contracts/policy/AdvancedPolicy.sol";
-import {AdvancedChecker} from "@excubiae/contracts/checker/AdvancedChecker.sol";
-import {Check} from "@excubiae/contracts/interfaces/IAdvancedChecker.sol";
-
-/// @title BaseERC721Policy.
-/// @notice Policy enforcer for Enclave Input validation.
-/// @dev Extends BasePolicy with Enclave specific checks.
-contract CRISPPolicy is AdvancedPolicy, IEnclavePolicy {
+import {BasePolicy} from "@excubiae/contracts/policy/BasePolicy.sol";
+import {BaseChecker} from "@excubiae/contracts/checker/BaseChecker.sol";
+import {ISemaphore} from "@semaphore-protocol/contracts/interfaces/ISemaphore.sol";
+
+/// @title CRISPPolicy
+/// @notice Policy contract for validating inputs based on Semaphore proofs and usage limits.
+contract CRISPPolicy is BasePolicy, IEnclavePolicy {
+    /// Errors
     error MainCalledTooManyTimes();
     error InvalidInitializationAddress();
+    error AlreadyEnforced();
 
+    /// State Variables
     uint8 public inputLimit;
     mapping(address subject => uint8 count) public enforced;
+    mapping(uint256 => bool) public spentNullifiers;
+
+    /// @notice Initializes the contract with appended bytes data for configuration.
+    /// @dev Decodes AdvancedChecker address and sets the owner.
+    function _initialize() internal virtual override {
+        bytes memory data = _getAppendedBytes();
+        (address sender, address baseCheckerAddr, uint8 _inputLimit) = abi
+            .decode(data, (address, address, uint8));
+        _transferOwnership(sender);
 
-    /// @notice Constructor to initialize the policy directly upon deployment.
-    /// @param _enclave The address of the Enclave contract that will call enforce (becomes the 'guarded' target).
-    /// @param _checker The address of the AdvancedChecker contract to use.
-    /// @param _inputLimit The maximum number of times enforce can be called per subject.
-    constructor(address _enclave, address _checker, uint8 _inputLimit) {
-        if (_checker == address(0) || _enclave == address(0))
-            revert InvalidInitializationAddress();
-        guarded = _enclave;
-        _transferOwnership(_enclave);
-        ADVANCED_CHECKER = AdvancedChecker(_checker);
-        SKIP_PRE = true;
-        SKIP_POST = true;
+        BASE_CHECKER = BaseChecker(baseCheckerAddr);
         inputLimit = _inputLimit;
     }
 
-    function _enforce(
+    /// @notice Validate the input and return the vote.
+    /// @param subject The subject to validate the policy on.
+    /// @param evidence Abi-encoded `ISemaphore.SemaphoreProof`.
+    function validate(
```

Comment:
hmm this could lead to unexpected errors since the `_enforce()` is already called by the [enforce()](https://github.com/privacy-scaling-explorations/excubiae/blob/main/packages/contracts/contracts/policy/BasePolicy.sol#L32) external fn and must be exposed / called only from there where all the logic is inherited from [BasePolicy](https://github.com/privacy-scaling-explorations/excubiae/blob/a7c8b6bc0ba998b13920a7e8044bf5fac22efe4b/packages/contracts/contracts/policy/BasePolicy.sol#L39) contract.

Please, double check this. If you need to return the vote, you can just store them or emit an event in the `internal` fn.

Reply:
I'm happy to sync anytime today to figure out a solution. The idea behind the `InputValidator` was to validate the proof and return the vote, that’s how it was originally set up. Since there might be multiple proof formats (e.g., some using Semaphore, some custom), we wanted to keep the input as generic bytes and let the validator handle decoding and validation. This led me to either introduce an intermediary contract (`InputValidator`) to handle `enforce`, or modify the policy to call `enforce` directly. I went with the latter to get things working quickly, we can now refine it together.